sklearn/tests/test_random_projection.py

```diff
@@ -60,18 +57,17 @@ def densify(matrix):
 # test on JL lemma
 ###############################################################################
 def test_invalid_jl_domain():
-    assert_raises(ValueError, johnson_lindenstrauss_min_dim, 100, eps=1.1)
-    assert_raises(ValueError, johnson_lindenstrauss_min_dim, 100, eps=0.0)
-    assert_raises(ValueError, johnson_lindenstrauss_min_dim, 100, eps=-0.1)
-    assert_raises(ValueError, johnson_lindenstrauss_min_dim, 0, eps=0.5)
+    with pytest.raises(ValueError):
+        johnson_lindenstrauss_min_dim(100, eps=1.1)
+        johnson_lindenstrauss_min_dim(100, eps=0.0)
+        johnson_lindenstrauss_min_dim(100, eps=-0.1)
+        johnson_lindenstrauss_min_dim(0, eps=0.5)
```

Comment:
This would only check the first call because it would error first. The other calls to `johnson_lindenstrauss_min_dim` would not run. Can adjust this test to use `pytest.mark.parametrize`?:

```python
@pytest.mark.parametrize("n_samples, eps", [
    (100, 1.1),
    (100, 0.0),
    (100, -0.1),
    (0, 0.5)
])
def test_invalid_jl_domain(n_samples, eps):
    with pytest.raises(ValueError):
        johnson_lindenstrauss_min_dim(n_samples, eps=eps)
```